clash-prelude/src/Clash/Explicit/Signal.hs

```diff
@@ -759,6 +762,35 @@ regEn = \clk rst gen initial en i ->
   register# clk rst (andEnable gen en) initial initial i
 {-# INLINE regEn #-}
 
+-- | A simple queuing FIFO that pushes data through whenever the
```

Comment:
This really is the most degenerate of FIFOs.... with many annoying properties;
1. Cannot distinguish new values from the initial values
2. Elements put into the FIFO cannot be gotten out until the FIFO has been completely filled up.

I would rather it be classified as a delay line rather than a FIFO.